iterm2 does not build on aarch64-darwin

### Describe the bug
The iterm2 package (https://search.nixos.org/packages?channel=21.05&show=iterm2&from=0&size=50&sort=relevance&type=packages&query=iterm2) does not build on aarch64-darwin. I'm getting the following error message:
```
warning: iTerm2 isn't code signed but requires entitlements. It is not possible to add entitlements to a binary without signing it. (in target 'iTerm2' from project 'iTerm2')
** BUILD FAILED **
```

### Steps To Reproduce
Steps to reproduce the behavior:
1. Try installing/building `iterm2`.
2. Observe the build failure.

### Expected behavior
The package to build.

### Screenshots
n/a

### Additional context
Complete logs are available here: https://gist.github.com/samuela/4bc3732dd27e2d6007ba0dc0390bd79a.

### Notify maintainers
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@tricktron

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 21.1.0, macOS 12.0.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.4`
 - channels(samuelainsworth): `"darwin, home-manager, nixpkgs-21.11pre330734.5cb226a06c4"`
 - channels(root): `"nixpkgs-21.11pre330734.5cb226a06c4"`
 - nixpkgs: `/Users/samuelainsworth/.nix-defexpr/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: iterm2
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/applications/terminal-emulators/iterm2/default.nix
+++ b/pkgs/applications/terminal-emulators/iterm2/default.nix
@@ -1,4 +1,4 @@
-{ lib, stdenv, fetchFromGitHub }:
+{ lib, stdenvNoCC, fetchFromGitHub }:
 
  /*
  This derivation is impure: it relies on an Xcode toolchain being installed
@@ -8,15 +8,15 @@
  option set to `relaxed` or `false`.
  */
 
-stdenv.mkDerivation rec {
+stdenvNoCC.mkDerivation rec {
   pname = "iterm2";
-  version = "3.4.0";
+  version = "3.4.13";
 
   src = fetchFromGitHub {
     owner = "gnachman";
     repo = "iTerm2";
     rev = "v${version}";
-    sha256 = "09nhrmi25zxw3vp0wlib9kjr3p1j6am2zpwimdzqn0c80fq1lwvi";
+    sha256 = "sha256-GOUdXBQCvM0oJ2/3zgKqDpfyCkHNwd1Qdopg5Mpyekg=";
   };
 
   patches = [ ./disable_updates.patch ];

